Adding C/C++ info to the docs (#648)
google/osv-scanner: docs/usage.md

osv-scanner --lockfile ':/path/to/my:projects/package-lock.json'
```

> OSV-Scanner also supports submoduled and vendored [C/C++ dependencies](#cc-scanning). There is no need to specify a lockfile, OSV-Scanner will automatically find them and return any corresponding vulnerabilities. 

Author: @oliverchang I put this in the "Specify lockfiles" section. Would you prefer it moved up closer to the list of lockfiles? Or is it ok here.

```bash
osv-scanner -L package-lock.json --output scan-results.txt
```

## C/C++ scanning

OSV-Scanner supports C/C++ projects. 

> Because the C/C++ ecosystem does not have a centralized package manager, C/C++ dependencies tend to be bundled with the project. Dependencies are either [submoduled](#submoduled-dependencies) or [vendored](#vendored-dependencies). In either case, OSV-Scanner is able to find known vulnerabilities in your project dependencies. 

Reviewer: Minor enhancement for clarity (I had to read this a couple of times to comprehend it Monday morning)

```suggestion
Because the C/C++ ecosystem does not have a centralized package manager, C/C++ dependencies tend to be bundled with the project's source. Dependencies are either [submoduled](#submoduled-dependencies) or [vendored](#vendored-dependencies). In either case, OSV-Scanner is able to find known vulnerabilities in your project dependencies. 
```

Author: @andrewpollock Maybe it should be "bundled with the project's source code"? Let me know what you think.

Reviewer: Yep, that's also fine by me.

> Because the C/C++ ecosystem does not have a centralized package manager, C/C++ dependencies tend to be bundled with the project. Dependencies are either [submoduled](#submoduled-dependencies) or [vendored](#vendored-dependencies). In either case, OSV-Scanner is able to find known vulnerabilities in your project dependencies. 

Author: ```suggestion
Because the C/C++ ecosystem does not have a centralized package manager, C/C++ dependencies tend to be bundled with the project's source code. Dependencies are either [submoduled](#submoduled-dependencies) or [vendored](#vendored-dependencies). In either case, OSV-Scanner is able to find known vulnerabilities in your project dependencies. 
```

> OSV-Scanner's C/C++ support is based on commit-level data. OSV's commit-level data covers the majority of C/C++ vulnerabilities within the OSV database, but users should be aware that there may be vulnerabilities in their dependencies that could be excluded from OSV-Scanner results. Adding more commit-level data to the database is an ongoing project. 

Reviewer: Clarification to head off potential questions about "why".
```suggestion
OSV-Scanner's C/C++ support is based on commit-level data. OSV's commit-level data covers the majority of C/C++ vulnerabilities within the OSV database, but users should be aware that there may be vulnerabilities in their dependencies that may not be in the OSV database and therefore not included in OSV-Scanner results. Adding more commit-level data to the database is an ongoing project, follow [#783](https://github.com/google/osv.dev/issues/783) for more details. 
```

Author: Thanks!

### Submoduled dependencies

> Submoduled dependencies are included in the project folder retain their Git histories. To scan a C/C++ project with submoduled dependencies:

Reviewer: Outlink for more details
```suggestion
[Submoduled](https://git-scm.com/book/en/v2/Git-Tools-Submodules) dependencies are included in the project's source retain their Git histories. To scan a C/C++ project with submoduled dependencies:
```

Author: @andrewpollock  same question as above. Should the term be "source code"?

Reviewer: Yes

> Submoduled dependencies are included in the project folder retain their Git histories. To scan a C/C++ project with submoduled dependencies:

Author: ```suggestion
[Submoduled](https://git-scm.com/book/en/v2/Git-Tools-Submodules) dependencies are included in the project's source code and retain their Git histories. To scan a C/C++ project with submoduled dependencies:
```

1. Navigate to the root folder of your project. 
2. Ensure that your submodules are up to date using `git submodule update`. 
3. Run scanner using `osv-scanner -r .`. 

### Vendored dependencies

> Vendored dependencies have been directly copied into the project folder, but do not retain their Git histories. OSV-Scanner uses OSV's [determineversion API](https://google.github.io/osv.dev/post-v1-determineversion/) to estimate each dependency's version (and associated Git Commit). Vulnerabilities for the estimated version are returned. This process requires no additional work from the user. Run OSV-Scanner as you normally would. 

Reviewer: ```suggestion
Vendored dependencies have been directly copied into the project folder, but do not retain their Git histories. OSV-Scanner uses OSV's [determineversion API](https://google.github.io/osv.dev/post-v1-determineversion/) to estimate each dependency's version (and associated Git commit). Vulnerabilities for the estimated version are returned. This process requires no additional work from the user. Run OSV-Scanner as you normally would. 
```

```bash
osv-scanner -L package-lock.json --output scan-results.txt
```

## C/C++ scanning

OSV-Scanner supports C/C++ projects. 

Because the C/C++ ecosystem does not have a centralized package manager, C/C++ dependencies tend to be bundled with the project. Dependencies are either [submoduled](#submoduled-dependencies) or [vendored](#vendored-dependencies). In either case, OSV-Scanner is able to find known vulnerabilities in your project dependencies. 

> 

Author: ```suggestion

OSV-Scanner's C/C++ support is based on commit-level data. OSV's commit-level data covers the majority of C/C++ vulnerabilities within the OSV database, but users should be aware that there may be vulnerabilities in their dependencies that could be excluded from OSV-Scanner results. Adding more commit-level data to the database is an ongoing project. 

```

> Because the C/C++ ecosystem does not have a centralized package manager, C/C++ dependencies tend to be bundled with the project. Dependencies are either [submoduled](#submoduled-dependencies) or [vendored](#vendored-dependencies). In either case, OSV-Scanner is able to find known vulnerabilities in your project dependencies. 

Author: I think this is a fair point. I took a stab at this. Let me know what you think.

> Because the C/C++ ecosystem does not have a centralized package manager, C/C++ dependencies tend to be bundled with the project. Dependencies are either [submoduled](#submoduled-dependencies) or [vendored](#vendored-dependencies). In either case, OSV-Scanner is able to find known vulnerabilities in your project dependencies. 

Reviewer: Yeah that covers the essence of the caveat I was trying to express, thanks!

### Submoduled dependencies

> Submoduled dependencies are included in the project folder retain their git histories. To scan a C/C++ project with submoduled dependencies:

Author: Agh. I will get used to this at some point. Thanks

1. Navigate to the root folder of your project. 
2. Ensure that your submodules are up to date using `git submodule update`. 

> 3. Run scanner using `osv-scanner .`. 

Author: I am not sure about the trailing slash, but `-r` makes sense to me.

> 3. Run scanner using `osv-scanner .`. 

Reviewer 2: A trailing slash is not needed.

 The -r flag is not necessarily needed, if we can find your vendor folder or git directory in the directory you passed in. Let's leave it in for now to be complete, it doesn't hurt, just takes longer to scan.

```bash
osv-scanner -L package-lock.json --output scan-results.txt
```

## C/C++ scanning

OSV-Scanner supports C/C++ projects. 

Because the C/C++ ecosystem does not have a centralized package manager, C/C++ dependencies tend to be bundled with the project. Dependencies are either [submoduled](#submoduled-dependencies) or [vendored](#vendored-dependencies). In either case, OSV-Scanner is able to find known vulnerabilities in your project dependencies. 

OSV-Scanner's C/C++ support is based on commit-level data. OSV's commit-level data covers the majority of C/C++ vulnerabilities within the OSV database, but users should be aware that there may be vulnerabilities in their dependencies that could be excluded from OSV-Scanner results. Adding more commit-level data to the database is an ongoing project. 

### Submoduled dependencies

Submoduled dependencies are included in the project folder retain their Git histories. To scan a C/C++ project with submoduled dependencies:

1. Navigate to the root folder of your project. 
2. Ensure that your submodules are up to date using `git submodule update`. 

> 3. Run scanner using `osv-scanner -r ./`. 

Reviewer: Seems like the trailing slash is not necessary

```suggestion
3. Run scanner using `osv-scanner -r .`. 
```
